<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ArduCopter Firmware Signing Tool</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGroupBox" name="connectionGroup">
      <property name="title">
       <string>Connection</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QComboBox" name="portComboBox"/>
       </item>
       <item>
        <widget class="QPushButton" name="refreshButton">
         <property name="text">
          <string>Refresh</string>
         </property>
         <property name="icon">
          <iconset resource="../resources.qrc">
           <normaloff>:/icons/refresh</normaloff>:/icons/refresh</iconset>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="baudRateComboBox">
         <item>
          <property name="text">
           <string>9600</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>19200</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>38400</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>57600</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>115200</string>
          </property>
         </item>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="autoConnectCheckBox">
         <property name="text">
          <string>Auto Connect</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="connectButton">
         <property name="text">
          <string>Connect</string>
         </property>
         <property name="icon">
          <iconset resource="../resources.qrc">
           <normaloff>:/icons/connect</normaloff>:/icons/connect</iconset>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="keyManagementGroup">
      <property name="title">
       <string>Key Management</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QPushButton" name="loadKeyButton">
         <property name="text">
          <string>Load Key from PFX</string>
         </property>
         <property name="icon">
          <iconset resource="../resources.qrc">
           <normaloff>:/icons/key</normaloff>:/icons/key</iconset>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="exportKeyButton">
         <property name="text">
          <string>Export Key</string>
         </property>
         <property name="icon">
          <iconset resource="../resources.qrc">
           <normaloff>:/icons/export</normaloff>:/icons/export</iconset>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="keyStatusLabel">
         <property name="text">
          <string>No key loaded</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="signingGroup">
      <property name="title">
       <string>Firmware Signing</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QPushButton" name="selectBinButton">
           <property name="text">
            <string>Select .bin File</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="binFilePath"/>
         </item>
         <item>
          <widget class="QPushButton" name="signBinButton">
           <property name="text">
            <string>Sign with Public Key</string>
           </property>
           <property name="icon">
            <iconset resource="../resources.qrc">
             <normaloff>:/icons/sign</normaloff>:/icons/sign</iconset>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QPushButton" name="selectApjButton">
           <property name="text">
            <string>Select .apj File</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="apjFilePath"/>
         </item>
         <item>
          <widget class="QPushButton" name="signApjButton">
           <property name="text">
            <string>Sign with Private Key</string>
           </property>
           <property name="icon">
            <iconset resource="../resources.qrc">
             <normaloff>:/icons/sign</normaloff>:/icons/sign</iconset>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QPushButton" name="verifyButton">
           <property name="text">
            <string>Verify Signature</string>
           </property>
           <property name="icon">
            <iconset resource="../resources.qrc">
             <normaloff>:/icons/verify</normaloff>:/icons/verify</iconset>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="hashCheckBox">
           <property name="text">
            <string>Check Hash</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="flashGroup">
      <property name="title">
       <string>DFU Flash</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_6">
       <item>
        <widget class="QPushButton" name="selectFirmwareButton">
         <property name="text">
          <string>Select Firmware</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="firmwarePath"/>
       </item>
       <item>
        <widget class="QPushButton" name="flashButton">
         <property name="text">
          <string>Flash Device</string>
         </property>
         <property name="icon">
          <iconset resource="../resources.qrc">
           <normaloff>:/icons/flash</normaloff>:/icons/flash</iconset>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QProgressBar" name="progressBar">
      <property name="value">
       <number>0</number>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="textVisible">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="statusLabel">
      <property name="text">
       <string>Ready</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources>
  <include location="../resources.qrc"/>
 </resources>
 <connections/>
</ui>
